src/main/resources/com/ncc/aif/aida_ontology.shacl

```diff
@@ -610,88 +610,110 @@ aida:SystemPropertyShape
    sh:path aida:system ;
    sh:class aida:System .
 
+# Shape to enforce properties shared by all justifications
+aida:SharedJustificationShape
+    a sh:NodeShape ;
 
-# Systems need to provide justifications from text, image, video, etc. for the assertions they make
-aida:TextJustificationShape
-   a sh:NodeShape ;
-   sh:targetClass aida:TextJustification ;
-   # each justification must have exactly one source document ID
-   sh:property [
-     sh:path aida:source ;
-     sh:datatype xsd:string ;
-     sh:minCount 1;
-     sh:maxCount 1 ];
-   # exactly one start character offset within the document
-   sh:property [
-      sh:path aida:startOffset ;
-      sh:or ( [sh:datatype xsd:int] [sh:datatype xsd:integer] )  ;
-      sh:minCount 1;
-      sh:maxCount 1 ];
-   # and exactly one inclusive end offset within the document
-   sh:property [
-       sh:path aida:endOffsetInclusive ;
-       sh:or ( [sh:datatype xsd:int] [sh:datatype xsd:integer] )  ;
-       sh:minCount 1;
-       sh:maxCount 1 ];
-   # an optional string label for user consumption (TODO: is this allowed? Probably not?)
-   sh:property [
-      sh:path skos:prefLabel ;
-      sh:datatype xsd:string ];
+    # each justification must have exactly one source document ID
```

Comment:
Suggest "...exactly one source document element ID".  The good news is that with `aida:SharedJustificationShape`, this only needs to be done once.  :)

Reply:
fixed